core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -327,6 +357,96 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
         }
     }
 
+    private long extractMeter(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToMeter(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract meter from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private long extractTons(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
+        String value = extractRoadAttribute(way, keys);
+
+        if (Helper.isEmpty(value)) return flags;
+
+        double val;
+        try {
+            val = stringToTons(value);
+        } catch (Throwable t) {
+            LOG.warn("Unable to extract tons from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
+            val = 0.0;
+        }
+
+        return valueEncoder.setDoubleValue(flags, val);
+    }
+
+    private String extractRoadAttribute(ReaderWay way, List<String> keys) {
+        String value = null;
+
+        for (String key : keys) {
+            value = way.getTag(key);
+            if (!Helper.isEmpty(value)) break;
+        }
+
+        return value;
+    }
+
+    double stringToTons(String value) {
```

Comment:
Wiki says it is tons ... but [here](https://karussell.wordpress.com/2015/11/01/units-in-openstreetmap/) is the real fun with units in OSM ;)

We can contribute our slightly more complex parsing methods from the commercial version.